src/components/Header.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { ActionNames, Errors, TodoContext } from './TodoContext';
+import { Todo } from '../types/Todo';
+import { USER_ID, createTodo } from '../api/todos';
+
+export const TEMP_USER_ID = -1;
+
+export const Header: React.FC = () => {
+  const { todos, dispatch, handleError } = useContext(TodoContext);
+  const [value, setValue] = useState('');
+  const [disabled, setDisabled] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    setValue(event.target.value);
+  };
+
+  const handleKeyDown = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (value.trim() === '') {
+      handleError(Errors.EmptyTitle);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: TEMP_USER_ID,
+      userId: USER_ID,
+      completed: false,
+      title: value.trim(),
+    };
+
+    setDisabled(true);
+
+    dispatch({ type: ActionNames.Add, payload: newTodo });
+
+    createTodo(newTodo)
+      .then(nt => {
+        dispatch({ type: ActionNames.Delete, payload: newTodo.id });
```

Comment:
It works, but let's do it as described in the task (with `tempTodo`) instead of adding and deleting this temporary todo in todos array